### PR TITLE
feat: add docker and docker compose

### PR DESCRIPTION
**Pull request by pandeyashu08**
This pr setups:
- [x] Docker
- [x] Docker Compose

Solves: #20 

Frontend is working, but the problem is with the backend.

### CODE REVIEW

File: frontend/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,18 @@
+FROM node:22-alpine
```
Comment (yesahem) [inline frontend/Dockerfile]: Run it over  node:20-alpine please

Comment (pandeyashu08) [reply frontend/Dockerfile]: @yesahem It doesn't matter it is independent of the OS. Whether you use alpine image or the debian image.

**Comment by pandeyashu08** [comment]
@yesahem Changed it to node 20 alpine

**Comment by yesahem** [comment]
> @yesahem Changed it to node 20 alpine

bro what about exposing the port in backend ?? @pandeyashu08 ,please check if it helps to get the access In local!

**Comment by pandeyashu08** [comment]
@yesahem It is directly taking it from the compose, I am giving the port there only. For frontend I was testing the image locally so that's why it was exposed there. Removing that will make no difference at all.

**Comment by yesahem** [comment]
Yeah that also makes sense to me 
Thanks !
Happy Coding ✨

**Pull request merged**